scripts/report_scripts/index.Rmd

```diff
@@ -159,48 +172,52 @@ get_h <- function() {
        fig.height*dpi/fig.retina)
 }
 
-# TODO change order or "variable" so that "WT" is always displayed on top
-# plot variant frequencies by location, no pooling
-melted <- reshape2::melt(approved_var_plot,
-                         id.vars = c("dates", "samplename", "location_name",
-                                     "coordinates_lat", "coordinates_long"))
-locations <- unique(melted$location_name)
-location_plot <- function (loc) {
-                   subset(melted, location_name == loc) %>%
-                       group_by(dates) %>%
-                       arrange(variable) %>%
-                       mutate (value = round (value, 4)) %>%
-                       plot_ly (type = 'scatter',
-                                mode = 'lines+markers',
-                                stackgroup = 'one',
+# TODO maybe skip the whole snippet and print plain warning in report
+if (nrow(approved_var_plot) > 0) { # check if filtered dataframe has actual values
+
+  # TODO change order or "variable" so that "WT" is always displayed on top
+  # plot variant frequencies by location, no pooling
+  melted <- reshape2::melt(approved_var_plot,
+                          id.vars = c("dates", "samplename", "location_name",
+                                      "coordinates_lat", "coordinates_long"))
+  locations <- unique(melted$location_name)
+  location_plot <- function (loc) {
+                    subset(melted, location_name == loc) %>%
+                        group_by(dates) %>%
+                        arrange(variable) %>%
+                        mutate (value = round (value, 4)) %>%
+                        plot_ly (type = 'scatter',
+                                  mode = 'lines+markers',
+                                  stackgroup = 'one',
+                                  showlegend = TRUE,
+                                  name = ~variable,
+                                  x = ~dates,
+                                  y = ~value,
+                                  color = ~variable,
+                                  colors = viridis_pal(option = "D")(6)) %>%
+                        layout (title = location,
                                 showlegend = TRUE,
-                                name = ~variable,
-                                x = ~dates,
-                                y = ~value,
-                                color = ~variable,
-                                colors = viridis_pal(option = "D")(6)) %>%
-                       layout (title = location,
-                               showlegend = TRUE,
-                               yaxis = list(title = "Frequency"))
+                                yaxis = list(title = "Frequency"))
+  }
+  plot_list <- list()
+  for ( i in seq_along(locations)) {
+    plot_name <- locations[i]
+    df <- paste0(locations[i],".df")
```

Comment:
I think it's bad style to modify df in a loop.  This makes code further down depend on this mutation, which can make it really hard to isolate behavior and test things.  Can we avoid mutating global state like this?